components/SearchBar/index.tsx

```diff
@@ -61,38 +73,55 @@ function Component() {
 
   const checkDomain = () => {
     const path = window.location.pathname.replace("/", "").toLowerCase();
-    const dom = path.split('.')[1];
-    if (
-      dom === 'tyron' || 'did' || 'vc' || 'treasury'
-    ) {
+    if (path.split('.')[1] === 'did' || path.split('.')[1] === 'vc' || path.split('.')[1] === 'treasury') {
       return true
     } else {
       return false
     }
   }
 
-  useEffect(() => {
-
-    //@todo review
+  const setUsername = () => {
     const path = window.location.pathname.replace("/", "").toLowerCase();
+    if (checkPath()) {
+      return path;
+    } else if (checkDomain()) {
+      return path.split('.')[0];
+    } else if (path.includes('.did') && path.includes('/')) {
+      return path.split('/')[0].split('.')[0]
+    } else if (path.split('/')[1] === 'did' || path.split('/')[1] === 'funds' || path.split('/')[1] === 'recovery') {
+      return path.split('/')[0]
```

Comment:
Here it could be username.did/funds so it needs that extra condition when there's the .did too?

Reply:
No, we dont need it. I already handle it with setUsername()